### PR TITLE
Remove ForegroundThreadAffinitizedObject usage

### PR DESCRIPTION
**Pull request by CyrusNajmabadi**

### CODE REVIEW

**Comment by genlu** [comment]
Why do we prefer using ThreadingContext directly over ForegroundThreadAffinitizedObject?

**Review by genlu — APPROVED** [review]

**Comment by Cosifne** [comment]
> Why do we prefer using ThreadingContext directly over ForegroundThreadAffinitizedObject?

Same question here : )

**Comment by jasonmalinowski** [comment]
Ditto on the question -- if we want to get rid of ForegroundThreadAffinitizedObject I won't mind too much, but if that's the direction we want to go we should excise it everywhere.

File: src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_SessionCancellingCommands.cs [hunk 1]
```diff
@@ -16,14 +17,14 @@ internal partial class EventHookupCommandHandler :
 
         public bool ExecuteCommand(EscapeKeyCommandArgs args, CommandExecutionContext context)
         {
-            AssertIsForeground();
+            Contract.ThrowIfFalse(_threadingContext.HasMainThread);
```
Comment (sharwell) [inline src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_SessionCancellingCommands.cs:20]: ❗ `HasMainThread` is a completely different check (it checks for the existence of a main thread). You want this:

```suggestion
            _threadingContext.ThrowIfNotOnUIThread();
```

**Review by sharwell — CHANGES_REQUESTED** [review]

**Comment by CyrusNajmabadi** [comment]
> Ditto on the question -- if we want to get rid of ForegroundThreadAffinitizedObject I won't mind too much, but if that's the direction we want to go we should excise it everywhere.

Workingon it.

**Comment by CyrusNajmabadi** [comment]
> Why do we prefer using ThreadingContext directly over ForegroundThreadAffinitizedObject?

So FTAO is a legacy thing we built prior to even having JTF and the nice APIs it has.  Today it's just a thin wrapper around IThreadingContext and really serves no good purpose.  So i'd prefer we just consistently use the platform compoennt here.

File: src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_SessionCancellingCommands.cs (hunk 1 above)
Comment (CyrusNajmabadi) [reply src/EditorFeatures/CSharp/EventHookup/EventHookupCommandHandler_SessionCancellingCommands.cs]: done.

File: src/EditorFeatures/CSharp/EventHookup/EventHookupSessionManager.cs [hunk 2]
```diff
@@ -16,12 +16,15 @@
...
 using Microsoft.VisualStudio.Text.Editor;
+using Roslyn.Utilities;
 
 namespace Microsoft.CodeAnalysis.Editor.CSharp.EventHookup
 {
     [Export]
-    internal sealed partial class EventHookupSessionManager : ForegroundThreadAffinitizedObject
+    internal sealed partial class EventHookupSessionManager
     {
+        public readonly IThreadingContext ThreadingContext;
```
Comment (sharwell) [inline src/EditorFeatures/CSharp/EventHookup/EventHookupSessionManager.cs:26]: Why public?

File: src/EditorFeatures/Core/GoToDefinition/AbstractGoToCommandHandler`2.cs [hunk 3]
```diff
@@ -94,7 +94,7 @@ public CommandState GetCommandState(TCommandArgs args)
     public bool ExecuteCommand(TCommandArgs args, CommandExecutionContext context)
     {
         // Should only be called on the UI thread.
```
Comment (sharwell) [inline src/EditorFeatures/Core/GoToDefinition/AbstractGoToCommandHandler`2.cs:96]:
```suggestion
```

File: src/EditorFeatures/Core/GoToDefinition/AbstractGoToCommandHandler`2.cs [hunk 4]
```diff
@@ -129,7 +129,7 @@ public bool ExecuteCommand(TCommandArgs args, CommandExecutionContext context)
         try
         {
             // Should only be called on the UI thread.
```
Comment (sharwell) [inline src/EditorFeatures/Core/GoToDefinition/AbstractGoToCommandHandler`2.cs:131]:
```suggestion
```

**Review by sharwell — APPROVED** [review]

**Pull request merged**